text/0000-deprecate-array-prototype-extensions.md

```diff
@@ -0,0 +1,130 @@
+
+We don't need to build anything new specifically, however, the bulk of the transition will be
+focused on deprecating the array prototype extensions.
+
+## How We Teach This
+
+An entry to the [Deprecation Guides](https://deprecations.emberjs.com/v4.x) will be added outlining the different recommended transition strategies.
+
+We will create a new lint rule `no-array-prototype-extensions` and set examples. Examples should have recommendations for equivalences.
+
+We will also create codemods to help with this migration.
+
+### Convenient methods
+Examples of deprecated and current code:
+```js
+import { set, action } from '@ember/object';
+import { uniqBy, sortBy } from 'lodash';
+
+export default class SampleComponent extends Component{
+  abc = ['x', 'y', 'z', 'x'];
+  
+  // deprecated
+  def = this.abc.without('x');
+  ghi = this.abc.uniq();
+  jkl = this.abc.toArray();
+  mno = this.abc.uniqBy('y');
+  pqr = this.abc.sortBy('z');
+  // ...
+
+  // current
+  def = this.abc.filter(el => el !== 'x');
+  ghi = [...new Set(this.abc)];
+  jkl = [...this.abc];
+  mno = uniqBy(this.abc, 'y');
+  pqr = sortBy(this.abc, 'z');
+};
+```
+
+### Observable based methods
+Examples of deprecated code:
+```js
+import { set, action } from '@ember/object';
+
+export default class SampleComponent extends Component{
+  abc = [];
+
+  @action
+  someAction(newItem) {
+    this.abc.pushObject('1');
+  }
+};
+```
+
+Examples of current code. 
+#### Option 1: tracked properties
+```js
+import { set, action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+export default class SampleComponent extends Component{
+  @tracked abc = [];
+
+  @action
+  someAction(newItem) {
+    this.abc = [...abc, newItem];
+  }
+};
+```
+
+#### Option 2: `TrackedArray`
+```js
+import { set, action } from '@ember/object';
+import { TrackedArray } from 'tracked-built-ins';
+
+export default class SampleComponent extends Component{
+  abc = new TrackedArray();
+
+  @action
+  someAction(newItem) {
+    abc.push(newItem);
+  }
+};
+```
+
+After the deprecated code is removed from Ember, we need to remove the [options to disable the array prototype extension](https://guides.emberjs.com/v4.2.0/configuring-ember/disabling-prototype-extensions/) from Official Guides and we also need to update the [Tracked Properties](https://guides.emberjs.com/v4.2.0/upgrading/current-edition/tracked-properties/#toc_arrays) `Arrays` section with updated suggestions.
+
+## Drawbacks
+- For users relying on Ember array prototype extensions, they will have to refactor their code and use equivalences appropriately.
+- A lot of existing Ember forums/blogs had been assuming the enabling of array prototype extensions which could cause confusions for users referencing them.
+
+
+## Alternatives
+- Continuing allowing array prototype extensions but turning the EXTEND_PROTOTYPES off by default.
```

Comment:
Based on https://github.com/emberjs/ember.js/issues/9269#issuecomment-250397695, addons have already turned the flag off by default. Though I couldn't find any official documentation for this.

Reply:
That's correct; addons have it off by default so that they don't opt apps in implicitly.